docs/src/main/asciidoc/central-logging.adoc

```diff
@@ -0,0 +1,377 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Quarkus - Central log management (Graylog, Logstash, Fluentd)
+
+include::./attributes.adoc[]
+
+This guide explains how you can send you log to a central logging system like Graylog, Logstash (inside the Elastic Stack or ELK - Elasticsearch, Logstash, Kibana) or
+Fluentd (inside EFK - Elasticsearch, Fluentd, Kibana).
+
+There is a lot of different way to centralize your logs (if you're using Kubernetes, the simplest way is to log to the console and ask you cluster administrator to integrate a central log manager inside your cluster).
+In this guide, we will expose how to send them to an external tool using the `quarkus-logging-gelf` extension that can use TCP or UDP to send logs in the Graylog Extented Log Format (GELF).
+
+The `quarkus-logging-gelf` extension will add a GELF log handler to the underlying logging backend that Quarkus uses (jboss-logmanager).
+By default, it is disabled, if you enable it but still use another handler (by default the console handler is enabled), your logs will be sent to both handlers.
+
+== Example application
+
+The following examples will all be based on the same example application that you can create based on the following steps:
+
+- Create an application with the `quarkus-logging-gelf` extension. You can use the following Maven command to create it:
+```
+mvn io.quarkus:quarkus-maven-plugin:1.0.0.Final:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=gelf-logging \
+    -DclassName="org.acme.quickstart.GelfLoggingResource" \
+    -Dpath="/gelf-logging" \
+    -Dextensions="logging-gelf"
+```
+
+- Add a log inside your rest endpoint
+```
+@Path("/gelf-logging")
+@ApplicationScoped
+public class GelfLoggingResource {
+    private static final Logger LOG = Logger.getLogger(GelfLogHandlerResource.class);
+
+    @GET
+    public void log() {
+        LOG.info("Some useful log message");
+    }
+
+}
+```
+
+- Configure the gelf log handler to send logs to an external UDP endpoint on the port 12201
+```
+quarkus.log.handler.gelf.enabled=true
+quarkus.log.handler.gelf.host=localhost
+quarkus.log.handler.gelf.port=12201
+```
+
+WARNING: UDP logging is currently not supported on native mode, use TCP if you plan to deploy your application as a native image.
```

Comment:
This might not be true anymore now that we're on GraalVM 19.3.0.

Reply:
On GraalVM 19.3 it didn't works anymore!

```
Exception in thread "main" java.lang.RuntimeException: Failed to start quarkus
	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:191)
	at io.quarkus.runtime.Application.start(Application.java:87)
	at io.quarkus.runtime.Application.run(Application.java:210)
	at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:41)
Caused by: java.lang.NullPointerException
	at java.net.InetAddress.getHostFromNameService(InetAddress.java:615)
	at java.net.InetAddress.getCanonicalHostName(InetAddress.java:589)
	at biz.paluch.logging.RuntimeContainer.isQualified(RuntimeContainer.java:148)
	at biz.paluch.logging.RuntimeContainer.getInetAddressWithHostname(RuntimeContainer.java:137)
	at biz.paluch.logging.RuntimeContainer.lookupHostname(RuntimeContainer.java:90)
	at biz.paluch.logging.RuntimeContainer.initialize(RuntimeContainer.java:67)
	at biz.paluch.logging.gelf.jul.GelfLogHandler.<init>(GelfLogHandler.java:54)
	at io.quarkus.logging.gelf.GelfLogHandlerRecorder.initializeHandler(GelfLogHandlerRecorder.java:17)
	at io.quarkus.deployment.steps.GelfLogHandlerProcessor$build28.deploy_0(GelfLogHandlerProcessor$build28.zig:76)
	at io.quarkus.deployment.steps.GelfLogHandlerProcessor$build28.deploy(GelfLogHandlerProcessor$build28.zig:36)
	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:68)
	... 3 more
```

Reply:
Interesting; any idea where the null reference is coming from?

Reply:
I use a workaround to configure the hostname resolver to localhost before network and it fixes the issue. 
I propose to open a followup issue to fix this properly after merging this, I suspect and issue on GraalVM side as the NPE is from code inside the JDK : `java.net.InetAddress.getHostFromNameService()`